### ISSUE TITLE
Infinite loop in pam_systemd_home.so

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#16813]
**systemd version the issue has been seen with**
> 246.2-2-arch

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > sshd not locking up

**Unexpected behaviour you saw**
 > sshd locking up in (pam_systemd_home.so) with 100% CPU load preventing remote login

**Steps to reproduce the problem**
None.

Ever since this update https://github.com/archlinux/svntogit-packages/commit/2d5af94ae55a5c98837ce9631f331ad2aad32bb3#diff-8d0411b338c83cd8cd8ad9d9db127101

I get occasional failures connecting over ssh to my machines. It doesn't happen predictably all the time, but out of 10 machines, I got 2 stuck with sshd in 100% CPU load and inability to log in remotely, just in the last week.

I tracked it down to infinite loop in pam_systemd_home.so loaded by the sshd process (I'm not using any of the new systemd home functionality - at least I didn't configure it in any way).

I get these errors in the log, which I presume mean I don't have any of the new systemd-home functionality enabled:
```
Aug 22 06:14:48 core sudo[2769199]: pam_systemd_home(sudo:account): Failed to query user record: Unit dbus-org.freedesktop.home1.service not found.
Aug 22 06:14:48 core dbus-daemon[5311]: [system] Activation via systemd failed for unit 'dbus-org.freedesktop.home1.service': Unit dbus-org.freedesktop.home1.service not found.
Aug 22 06:14:48 core dbus-daemon[5311]: [system] Activating via systemd: service name='org.freedesktop.home1' unit='dbus-org.freedesktop.home1.service' requested by ':1.6564' (uid=0 pid=2769199 comm="sudo su ")
```

The only thing I managed to discern from the gdb attaching to the sshd process is that loop contains a strcmp for `org.freedesktop.DBus.Error.NoReply` error.

**Comment by megous** [comment]
Does this possibly fix the 100% load and blocking of sshd? The error message I posted only happens when I attach a display to my board and run some sudo command. It's probably unrelated to sshd issue.

**Comment by megous** [comment]
Anyway, I disabled pam_systemd_home.so in pam.d again and didn't see any lockup with 100% CPU load in days.

I tried looking at pam_systemd_home.so, but it seems too complicated, and clearly not something I'd like to be loading into the sshd process anyway.

Issue is also with openssh, as it doesn't seem to have any timeout mechanism when waiting for the forked PAM auth process to succeed or fail, so when pam_systemd_home.so locks up in an infinite loop, sshd will just wait for it indefinitely and block further login attempts.
